website/src/lib/oasst_inference_client.ts

```diff
@@ -1,69 +1,38 @@
-/* eslint-disable @typescript-eslint/no-explicit-any */
 import axios, { AxiosRequestConfig } from "axios";
-import Cookies from "cookies";
-import type { NextApiRequest, NextApiResponse } from "next";
-import { JWT } from "next-auth/jwt";
-import { ChatItem, InferenceTokenResponse, InferenceMessage, InferencePostMessageResponse } from "src/types/Chat";
+import type { JWT } from "next-auth/jwt";
+import { ChatItem, InferenceMessage, InferencePostMessageResponse } from "src/types/Chat";
+import type { Readable } from "stream";
 
-// TODO: this class could be structured better
 export class OasstInferenceClient {
-  private readonly cookies: Cookies;
-  private inferenceToken: string;
-  private readonly userTokenSub: string;
+  // this is not a long lived class, this is why the token is immutable
+  constructor(private readonly inferenceAccessToken: string) {}
 
-  constructor(req: NextApiRequest, res: NextApiResponse, token: JWT) {
-    this.cookies = new Cookies(req, res);
-    this.inferenceToken = this.cookies.get("inference_token");
-    this.userTokenSub = token.sub;
-  }
-
-  async request<T = any>(method: "GET" | "POST" | "PUT" | "DELETE", path: string, init?: AxiosRequestConfig) {
-    const token = await this.get_token();
-    const { data } = await axios<T>(process.env.INFERENCE_SERVER_HOST + path, {
-      method,
+  async request<T = unknown>(path: string, init?: AxiosRequestConfig) {
+    const { data } = await axios<T>(process.env.NEXT_PUBLIC_INFERENCE_SERVER_HOST + path, {
```

Comment:
[we have to configure the build pipeline for the variable to be available](https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser). I will do it when back home

Reply:
@notmd I went with your suggestion of passing it as a server side prop, I was hoping to have a more "global" solution, but since we don't have to pass the variable many levels deep, I think it is okay for now.